src/main/kotlin/no/nav/tiltakspenger/datadeling/motta/infra/db/MottaNyBehandlingPostgresRepo.kt

```diff
@@ -0,0 +1,189 @@
+package no.nav.tiltakspenger.datadeling.motta.infra.db
+
+import arrow.core.Either
+import arrow.core.right
+import kotlinx.serialization.json.Json
+import kotliquery.queryOf
+import mu.KotlinLogging
+import no.nav.tiltakspenger.datadeling.motta.app.KunneIkkeLagreBehandling
+import no.nav.tiltakspenger.datadeling.motta.app.MottaNyBehandlingRepo
+import no.nav.tiltakspenger.libs.common.Fnr
+import no.nav.tiltakspenger.libs.periodisering.Periode
+import no.nav.tiltakspenger.libs.persistering.infrastruktur.PostgresSessionFactory
+
+internal class MottaNyBehandlingPostgresRepo(
+    private val sessionFactory: PostgresSessionFactory,
+) : MottaNyBehandlingRepo {
+    val log = KotlinLogging.logger { }
+
+    override fun lagre(behandling: TiltakspengerBehandling): Either<KunneIkkeLagreBehandling, Unit> {
+        log.info { "Lagrer behandling med id ${behandling.behandlingId}" }
+        if (eksisterer(behandling.behandlingId)) {
+            log.info { "Behandling med id = ${behandling.behandlingId} finnes allerede. Oppdaterer med ny info" }
+            oppdaterBehandling(behandling, sessionFactory)
+        } else {
+            opprettBehandling(behandling, sessionFactory)
+            log.info { "Behandling med id ${behandling.behandlingId} lagret." }
+        }
+
+        return Unit.right()
+    }
+
+    private fun oppdaterBehandling(
+        behandling: TiltakspengerBehandling,
+        sessionFactory: PostgresSessionFactory,
+    ) {
+        sessionFactory.withSession { session ->
+            session.run(
+                // TODO Kew: Kanskje ikke godta at man kan endre fnr, søknad_journalpost_id etc?
```

Comment:
Generelt synes jeg ikke databaselaget skal overprøve domene på slike ting.

For denne applikasjonen (tiltakspenger-datadeling) skal den ikke overprøve tiltakspenger-saksbehandling-api. Så her moser vi bare alt rett inn!